src/pysdmx/model/message.py

```diff
@@ -34,3 +37,80 @@ class Header(Struct, frozen=True, kw_only=True):
     receiver: Optional[str] = None
     source: Optional[str] = None
     dataset_action: Optional[ActionType] = None
+
+
+ORGS = "OrganisationSchemes"
+CLS = "Codelists"
+CONCEPTS = "ConceptSchemes"
+
+MSG_CONTENT_PKG = {
+    ORGS: ItemScheme,
+    CLS: Codelist,
+    CONCEPTS: ConceptScheme,
+}
+
+
+class Message(Struct, frozen=True):
+    """Message class holds the content of SDMX Message.
+
+    Attributes:
+        content (Dict[str, Any]): Content of the message. The keys are the
+            content type (e.g. ``OrganisationSchemes``, ``Codelists``, etc.),
+            and the values are the content objects (e.g. ``ItemScheme``,
+            ``Codelist``, etc.).
+    """
+
+    content: Dict[str, Any]
+
+    def __post_init__(self) -> None:
+        """Checks if the content is valid."""
+        for content_key, content_value in self.content.items():
+            if content_key not in MSG_CONTENT_PKG:
+                raise ValueError(f"Invalid content type: {content_key}")
+
+            for obj_ in content_value.values():
+                if not isinstance(obj_, MSG_CONTENT_PKG[content_key]):
+                    raise ValueError(
+                        f"Invalid content value type: {type(obj_).__name__} "
+                        f"for {content_key}"
+                    )
+
+    def __get_elements(self, type_: str) -> Dict[str, Any]:
+        """Returns the elements from content."""
+        if type_ in self.content:
+            return self.content[type_]
+        raise ValueError(f"No {type_} found in content")
+
+    def __get_element_by_uid(self, type_: str, unique_id: str) -> Any:
+        """Returns a specific element from content."""
+        if type_ not in self.content:
+            raise ValueError(f"No {type_} found")
+
+        if unique_id in self.content[type_]:
+            return self.content[type_][unique_id]
+
+        raise ValueError(f"No {type_} with id {unique_id} found in content")
```

Comment:
I don't know how this is being used but, if not having something of a certain type is acceptable, we could return an empty collection, instead of throwing an error. 

If it is not acceptable and so an error must be returned, then maybe we should return the NotFound error provided by the pysdmx `errors` module?

Reply:
The goal is to provide a method to extract a specific element from the Message content. I do not have a strong option to return an empty dictionary or an Error. I do prefer an error for better usability.

I will implement it using the NotFound error, do you agree? @sosna

Reply:
Sure, that sounds good to me. Thanks, @javihern98